src/game/client/draw/drawable.cpp

```diff
@@ -985,3 +1002,826 @@ void Drawable::Draw_UI_Text()
+        } else if (object_id != OBJECT_UNK) {
+            Object *object = g_theGameLogic->Find_Object_By_ID(object_id);
+            captainslog_relassert(object_id == OBJECT_UNK,
+                6,
+                "Drawable::xfer - Drawable '%s' is not attached to an object but should be attached to object '%s' with id "
+                "'%d'",
+                Get_Template()->Get_Name().Str(),
+                object != nullptr ? object->Get_Template()->Get_Name().Str() : "Unknown",
+                object_id);
+        }
+    }
+
+    xfer->xferUnsignedInt(&m_status);
+    xfer->xferUnsignedInt(&m_drawBits);
+    xfer->xferUnsignedInt(&m_previousDrawBits);
+    xfer->xferUser(&m_fadingMode, sizeof(m_fadingMode));
+    xfer->xferUnsignedInt(&m_curFadeFrame);
+    xfer->xferUnsignedInt(&m_timeToFade);
+
+    bool loco = m_drawableLocoInfo != nullptr;
+    xfer->xferBool(&loco);
+
+    if (loco) {
+        if (xfer->Get_Mode() == XFER_LOAD && m_drawableLocoInfo == nullptr) {
+            m_drawableLocoInfo = new DrawableLocoInfo();
+        }
+
+        xfer->xferReal(&m_drawableLocoInfo->m_pitch);
+        xfer->xferReal(&m_drawableLocoInfo->m_pitchRate);
+        xfer->xferReal(&m_drawableLocoInfo->m_roll);
+        xfer->xferReal(&m_drawableLocoInfo->m_rollRate);
+        xfer->xferReal(&m_drawableLocoInfo->m_thrustWobble);
+        xfer->xferReal(&m_drawableLocoInfo->m_accelerationPitch);
+        xfer->xferReal(&m_drawableLocoInfo->m_accelerationPitchRate);
+        xfer->xferReal(&m_drawableLocoInfo->m_accelerationRoll);
+        xfer->xferReal(&m_drawableLocoInfo->m_accelerationRollRate);
+        xfer->xferReal(&m_drawableLocoInfo->m_overlapZVel);
+        xfer->xferReal(&m_drawableLocoInfo->m_overlapZ);
+        xfer->xferReal(&m_drawableLocoInfo->m_thrust);
+        xfer->xferReal(&m_drawableLocoInfo->m_wheelInfo.m_frontLeftHeightOffset);
+        xfer->xferReal(&m_drawableLocoInfo->m_wheelInfo.m_frontRightHeightOffset);
+        xfer->xferReal(&m_drawableLocoInfo->m_wheelInfo.m_rearLeftHeightOffset);
+        xfer->xferReal(&m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset);
+        xfer->xferReal(&m_drawableLocoInfo->m_wheelInfo.m_wheelAngle);
+        xfer->xferInt(&m_drawableLocoInfo->m_wheelInfo.m_framesAirborneCounter);
+        xfer->xferInt(&m_drawableLocoInfo->m_wheelInfo.m_framesAirborne);
+    }
+
+    Xfer_Drawable_Modules(xfer);
+    xfer->xferUser(&m_stealthLook, sizeof(m_stealthLook));
+    xfer->xferInt(&m_flashTime);
+    xfer->xferColor(&m_flashColor);
+    xfer->xferBool(&m_hidden);
+    xfer->xferBool(&m_stealthInvisible);
+    xfer->xferReal(&m_stealthEmissiveScale);
+    xfer->xferBool(&m_instanceIsIdentity);
+    xfer->xferUser(&m_instance, sizeof(m_instance));
+    xfer->xferReal(&m_instanceScale);
+    xfer->xferObjectID(&m_drawableInfo.object_id);
+
+    if (version < 2) {
+        captainslog_dbgassert(xfer->Get_Mode() == XFER_LOAD, "Drawable::xfer - Writing an old format!!!");
```

Comment:
Better: `xfer->Get_Mode() != XFER_WRITE`, because XferType has 4 values.

Reply:
No this is correct to the original code (the original code does "if Get_Mode != XFER_LOAD, print assert".

Reply:
I understand. But the assert condition is wrong. You can write it like I suggested and mark it as "BUGFIX".

Reply:
No the assert condition is correct, the assert is supposed to trip anytime xfer mode is not XFER_LOAD, there is no bug here.
The intent of the developers is absolutely that the assert should trip if xfer mode is XFER_CRC or XFER_INVALID or anything else. (which makes sense given how and when this code runs)